spring-data-jdbc/src/main/java/org/springframework/data/jdbc/repository/query/PartTreeJdbcQuery.java

```diff
@@ -192,7 +192,7 @@ private JdbcQueryExecution<?> getJdbcQueryExecution(@Nullable ResultSetExtractor
 			if (getQueryMethod().isModifyingQuery()) {
 				return createModifyingQueryExecutor();
 			} else {
-				return createReadingQueryExecution(extractor, rowMapper);
+				return createReadingQueryExecution(extractor, () -> rowMapper);
```

Comment:
Shouldn't we replace the parameter `rowMapper` with a `Supplier`? This method is called a single time and the `rowMapper` is constructed right before that usage so converting it to a `Supplier` should be fairly simple.